File: Problems/977_Squares_of_a_Sorted_Array.py
from typing import List

class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left = 0
        right = len(nums)-1
        index = right
        res = [0]*len(nums)
        while left <= right:
            if abs(nums[left])>abs(nums[right]):
                res[index] = nums[left]**2
                left += 1
            else:
                res[index] = nums[right]**2
                right -= 1
            index -= 1
        return res

File: Problems/977_Squares_of_a_Sorted_Array.cpp
#include<vector>
#include<math.h>

using namespace std;
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int left  = 0;
        int right = nums.size()-1;
        int index = nums.size()-1;

        vector<int> res(nums.size(), 0);
        while(left<=right){
            if(abs(nums[left]) > abs(nums[right])){
                res[index] = pow(nums[left], 2);
                ++left;
            }else{
                res[index] = pow(nums[right], 2);
                --right;
            }
            --index;
        }
        return res;
    }
};